tests/query_api.rs

```diff
@@ -125,6 +125,19 @@ fn query_read_entity_data() {
     assert_eq!(components.len(), count);
 }
 
+#[test]
+fn query_try_read_entity_data() {
+    let _ = env_logger::builder().is_test(true).try_init();
+
+    let universe = Universe::new();
+    let mut world = universe.create_world();
+    world.insert((), Some((Pos(1., 2., 3.),)));
+    world.insert((), Some((Pos(4., 5., 6.), Rot(0.4, 0.5, 0.6))));
+
+    let mut query = TryRead::<Rot>::query();
+    assert_eq!(query.iter(&world).map(|x| x.map(|x| *x)).collect::<Vec<_>>(), &[None, Some(Rot(0.4, 0.5, 0.6))]);
```

Comment:
This test shouldn't assume entity order. In practise this will work, but ordering is not something legion guarentees in queries. For example, if you inserted another entity with only a `Pos`, it would end up 2nd in the query because it will be inserted into the first chunk. Entities are also swap-removed, so deletions and component layout changes will shuffle the entities in a chunk.

Reply:
Whoops, good catch, thanks! Fixed.